Documentation/contributors/design-proposals/webhooks.md

```diff
@@ -0,0 +1,213 @@
+        - serviceAccountName: my-operator
+          rules:
+          - apiGroups:
+            - "*"
+            ...
+      permissions:
+        - serviceAccountName: my-operator
+          rules:
+          - apiGroups:
+            - ""
+            ...
+      deployments:
+        - name: webhook-deployment
+          spec:
+            replicas: 1
+            selector:
+              matchLabels:
+                name: mywebhook
+            template:
+              metadata:
+                labels:
+                  name: mywebhook
+              spec:
+                serviceAccountName: my-operator
+                containers:
+                - name: webhook
+                  image: operator-framework/fake-webhook-image:latest
+                  imagePullPolicy: IfNotPresent
+                  ports:
+                  - containerPort: 443
+                  env:
+                  - name: NAMESPACE
+                    valueFrom:
+                      fieldRef:
+                        fieldPath: metadata.annotations['olm.targetNamespaces']
+                  volumeMounts:
+                    - mountPath: /keys
+                      name: tls-keys
+                      readonly: true
+                - name: operator
+                  image: operator-framework/fake-operator:latest
+                  imagePullPolicy: IfNotPresent
+                  env:
+                  - name: NAMESPACE
+                    valueFrom:
+                      fieldRef:
+                        fieldPath: metadata.annotations['olm.targetNamespaces']
+                  - name: MY_POD_NAME
+                    valueFrom:
+                      fieldRef:
+                        fieldPath: metadata.name
+                volumes:
+                - name: tls-keys
+                    secretName: service-tls
+      services:
+      - name: mywebhook-service
+        namespace: test-ns
+        spec:
+          selector:
+            name: mywebhook
+          ports:
+          - protocol: TCP
+            port 443
+            targetPort: 443
+      validatingWebhooks: # also available, mutatingWebhooks
+      - name: olm-admission
+        webhooks:
+          - name: csv.operators.olm.com
+          rules:
+            - apiGroups:
+              - "operators.olm.com"
+              apiVersions:
+              - v1alpha1
+              operations:
+              - CREATE
+              resources:
+              - clusterserviceversions
+          clientConfig:
+            service:
+              name: mywebhook-service
+              namespace: test-ns
+              path: "/admit"
+
+Note that the CABundle field in the clientConfig is set to use the general CA from the Kubernetes apiserver.
+
+### Description for v2
+
+The following example is a different take which would remove control from the user and have OLM manage the missing bits that aren't specified. The missing fields and objects would be created by OLM on the fly.
```

Comment:
"on the fly" - I'm not quite sure how. I was originally thinking of this sort of thing happening during resolving, but in out of band discussions adding data to the bundles has been mentioned which I don't fully understand what that'd look like.

Reply:
This would be handled during CSV creation, similar to how we handle apiservices. See: https://github.com/operator-framework/operator-lifecycle-manager/blob/master/pkg/controller/operators/olm/operator.go#L1031-L1036

@njhale also made the point that we're not using CSRs there, but we probably should be